lib/rules/no-restricted-properties.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @fileoverview Rule to disallow certain object properties
+ * @author Will Klein & Eli White
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow certain properties on certain objects",
+            category: "Node.js and CommonJS",
+            recommended: false
+        },
+
+        schema: {
+            type: "array",
+            items: {
+                type: "object",
+                properties: {
+                    object: {
+                        type: "string"
+                    },
+                    property: {
+                        type: "string"
+                    },
+                    message: {
+                        type: "string"
+                    }
+                },
+                additionalProperties: false,
+                required: [
+                    "object",
+                    "property"
+                ]
+            },
+            uniqueItems: true
+        }
+    },
+
+    create(context) {
+        const restrictedCalls = context.options;
+
+        if (restrictedCalls.length === 0) {
+            return {};
+        }
+
+        const restrictedProperties = restrictedCalls.reduce(function(restrictions, option) {
+            const objectName = option.object;
+            const propertyName = option.property;
+
+            if (!restrictions.has(objectName)) {
+                restrictions.set(objectName, new Map());
+            }
+
+            restrictions.get(objectName).set(propertyName, {
+                message: option.message
+            });
+
+            return restrictions;
+        }, new Map());
+
+        return {
+            MemberExpression(node) {
+                const objectName = node.object && node.object.name;
+                const propertyName = node.property && node.property.name;
```

Comment:
I think @mysticatea had recently added a "get simple property name" method
to ast-utils, might want to hunt that down.

On Sep 2, 2016 8:56 AM, "Ilya Volodin" <notifications@github.com> wrote:

> In lib/rules/no-restricted-properties.js
> <https://github.com/eslint/eslint/pull/7017#discussion_r77349937>:
>
> > +
> > +            if (!restrictions.has(objectName)) {
> > +                restrictions.set(objectName, new Map());
> > +            }
> > +
> > +            restrictions.get(objectName).set(propertyName, {
> > +                message: option.message
> > +            });
> > +
> > +            return restrictions;
> > +        }, new Map());
> > +
> > +        return {
> > +            MemberExpression(node) {
> > +                const objectName = node.object && node.object.name;
> > +                const propertyName = node.property && node.property.name;
>
> This is only going to catch foo.bar notation. Should we also catch
> foo['bar'] as well?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/eslint/eslint/pull/7017/files/576e4cc27d1d2f01961be20a41fa94df19a75b19#r77349937>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARWekx4AnK_LRONX9DtdWUi0el5Plroks5qmCsngaJpZM4JwEv3>
> .
>